### ISSUE TITLE
ClassNotFoundExcption for RemovalCause on Forge-based servers

### ISSUE THREAD

**Issue by LiskoSlayer63** [BlueMap-Minecraft/BlueMap#231]
### What i did / Steps to reproduce
Put the `bluemap.jar` to mods folder and start Mohist.

### Expected result
Map to work normally

### Actual result
Multiple ClassNotFound errors ([log here](https://mclo.gs/EUX5CA0))

### Context
**BlueMap Version:**  
`1.5.5-forge-1.16.4` and `1.6.0-forge-1.16.4`
**Mohist Version:**  
`1.16.5-750`

Using Windows Server with OpenJDK 11

**Comment by TBlueF** [comment]
This error mostly happens on Forge. It's completely harmless, only annoying ^^

<details><summary>Log</summary>

```
[03:50:41] [Thread-28/INFO] [BlueMap/]: Loaded!
[03:50:47] [ForkJoinPool-5-worker-23/ERROR] [net.minecraftforge.eventbus.EventSubclassTransformer/EVENTBUS]: Could not find parent de/bluecolored/shadow/benmanes/caffeine/cache/RemovalCause for class de/bluecolored/shadow/benmanes/caffeine/cache/RemovalCause$1 in classloader jdk.internal.loader.ClassLoaders$AppClassLoader@46cdf8bd on thread Thread[ForkJoinPool-5-worker-23,5,SERVER]
[03:50:47] [ForkJoinPool-5-worker-23/ERROR] [net.minecraftforge.eventbus.EventSubclassTransformer/EVENTBUS]: An error occurred building event handler
java.lang.ClassNotFoundException: de.bluecolored.shadow.benmanes.caffeine.cache.RemovalCause
at jdk.internal.loader.BuiltinClassLoader.loadClass(Unknown Source) ~[?:?]
at jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(Unknown Source) ~[?:?]
at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:?]
at net.minecraftforge.eventbus.EventSubclassTransformer.buildEvents(EventSubclassTransformer.java:62) ~[eventbus-4.0.0.jar:?]
at net.minecraftforge.eventbus.EventSubclassTransformer.transform(EventSubclassTransformer.java:44) [eventbus-4.0.0.jar:?]
at net.minecraftforge.eventbus.EventBusEngine.processClass(EventBusEngine.java:21) [eventbus-4.0.0.jar:?]
at net.minecraftforge.eventbus.service.ModLauncherService.processClassWithFlags(ModLauncherService.java:20) [eventbus-4.0.0.jar:4.0.0+65+master.6080a3f]
at cpw.mods.modlauncher.LaunchPluginHandler.offerClassNodeToPlugins(LaunchPluginHandler.java:85) [modlauncher-8.0.9.jar:?]
at cpw.mods.modlauncher.ClassTransformer.transform(ClassTransformer.java:120) [modlauncher-8.0.9.jar:?]
at cpw.mods.modlauncher.TransformingClassLoader$DelegatedClassLoader.findClass(TransformingClassLoader.java:265) [modlauncher-8.0.9.jar:?]
at cpw.mods.modlauncher.TransformingClassLoader.loadClass(TransformingClassLoader.java:136) [modlauncher-8.0.9.jar:?]
at cpw.mods.modlauncher.TransformingClassLoader.loadClass(TransformingClassLoader.java:98) [modlauncher-8.0.9.jar:?]
at java.lang.ClassLoader.loadClass(Unknown Source) [?:?]
at de.bluecolored.shadow.benmanes.caffeine.cache.BoundedLocalCache.evictFromMain(BoundedLocalCache.java:747) [bluemap:?]
at de.bluecolored.shadow.benmanes.caffeine.cache.BoundedLocalCache.evictEntries(BoundedLocalCache.java:612) [bluemap:?]
at de.bluecolored.shadow.benmanes.caffeine.cache.BoundedLocalCache.maintenance(BoundedLocalCache.java:1480) [bluemap:?]
at de.bluecolored.shadow.benmanes.caffeine.cache.BoundedLocalCache.performCleanUp(BoundedLocalCache.java:1448) [bluemap:?]
at de.bluecolored.shadow.benmanes.caffeine.cache.BoundedLocalCache$PerformCleanupTask.run(BoundedLocalCache.java:3293) [bluemap:?]
at de.bluecolored.shadow.benmanes.caffeine.cache.BoundedLocalCache$PerformCleanupTask.exec(BoundedLocalCache.java:3280) [bluemap:?]
at java.util.concurrent.ForkJoinTask.doExec(Unknown Source) [?:?]
at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source) [?:?]
at java.util.concurrent.ForkJoinPool.scan(Unknown Source) [?:?]
at java.util.concurrent.ForkJoinPool.runWorker(Unknown Source) [?:?]
at java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source) [?:?]
```

</details>

**Comment by LiskoSlayer63** [comment]
Well, I don't know if it's related to those errors or not, but the map itself doesn't work. It just stays black. I'm using the CLI version at the moment which works fine.

**Comment by TBlueF** [comment]
@LiskoSlayer63 definitely not related :)
You can try this: https://bluemap.bluecolored.de/wiki/FAQ.html#q-i-have-a-black-map--a-lot-of-the-map-is-missing
Or feel free to join our discord if you want help with that :)

**Comment by LiskoSlayer63** [comment]
Already tried that, it also doesn't generate the folders correctly on startup. I'm starting to think that this is a Mohist problem because the commands aren't working either. It loads, tho.

**Comment by TBlueF** [comment]
This is not a good place for support :)

**Comment by LiskoSlayer63** [comment]
Yeah I just reported a bug, but seems like I need to report it to Mohist instead.

**Comment by TBlueF** [comment]
This **should** be fixed in 1.6.1, but it is really hard for me to reproduce this reliably so i might be wrong.
Closing this for now, but **please reopen this if you still have this issue in 1.6.1+**! :)

**Comment by LiskoSlayer63** [comment]
> 
> 
> This **should** be fixed in 1.6.1, but it is really hard for me to reproduce this reliably so i might be wrong.
> Closing this for now, but **please reopen this if you still have this issue in 1.6.1+**! :)

I'm sorry to depress you, but the error spam is still there when using 1.6.1. Didn't try the 1.6.2 yet, because I need to update the front-end too...


**EDIT:**
Have no idea how to reopen an issue.

**Comment by TBlueF** [comment]
Thanks for testing it, i'll keep trying ^^'

**Comment by MarcelCoding** [comment]
The error is still present:
```
[16:43:25] [ForkJoinPool-5-worker-5/ERROR] [ne.mi.ev.EventSubclassTransformer/EVENTBUS]: Could not find parent de/bluecolored/shadow/benmanes/caffeine/cache/RemovalCause for class de/bluecolored/shadow/benmanes/caffeine/cache/RemovalCause$5 in classloader jdk.internal.loader.ClassLoaders$AppClassLoader@5cb0d902 on thread Thread[ForkJoinPool-5-worker-5,5,SERVER]
[16:43:25] [ForkJoinPool-5-worker-5/ERROR] [ne.mi.ev.EventSubclassTransformer/EVENTBUS]: An error occurred building event handler
java.lang.ClassNotFoundException: de.bluecolored.shadow.benmanes.caffeine.cache.RemovalCause
        at jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581) ~[?:?] {}
        at jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178) ~[?:?] {}
        at java.lang.ClassLoader.loadClass(ClassLoader.java:522) ~[?:?] {}
        at net.minecraftforge.eventbus.EventSubclassTransformer.buildEvents(EventSubclassTransformer.java:62) ~[eventbus-4.0.0.jar:?] {}
        at net.minecraftforge.eventbus.EventSubclassTransformer.transform(EventSubclassTransformer.java:44) [eventbus-4.0.0.jar:?] {}
        at net.minecraftforge.eventbus.EventBusEngine.processClass(EventBusEngine.java:21) [eventbus-4.0.0.jar:?] {}
        at net.minecraftforge.eventbus.service.ModLauncherService.processClassWithFlags(ModLauncherService.java:20) [eventbus-4.0.0.jar:4.0.0+65+master.6080a3f] {}
        at cpw.mods.modlauncher.LaunchPluginHandler.offerClassNodeToPlugins(LaunchPluginHandler.java:85) [modlauncher-8.1.3.jar:?] {}
        at cpw.mods.modlauncher.ClassTransformer.transform(ClassTransformer.java:120) [modlauncher-8.1.3.jar:?] {}
        at cpw.mods.modlauncher.TransformingClassLoader$DelegatedClassLoader.findClass(TransformingClassLoader.java:265) [modlauncher-8.1.3.jar:?] {}
        at cpw.mods.modlauncher.TransformingClassLoader.loadClass(TransformingClassLoader.java:136) [modlauncher-8.1.3.jar:?] {re:classloading}
        at cpw.mods.modlauncher.TransformingClassLoader.loadClass(TransformingClassLoader.java:98) [modlauncher-8.1.3.jar:?] {re:classloading}
        at java.lang.ClassLoader.loadClass(ClassLoader.java:522) [?:?] {}
        at de.bluecolored.shadow.benmanes.caffeine.cache.BoundedLocalCache.evictFromMain(BoundedLocalCache.java:742) [bluemap:?] {re:classloading}
        at de.bluecolored.shadow.benmanes.caffeine.cache.BoundedLocalCache.evictEntries(BoundedLocalCache.java:612) [bluemap:?] {re:classloading}
        at de.bluecolored.shadow.benmanes.caffeine.cache.BoundedLocalCache.maintenance(BoundedLocalCache.java:1480) [bluemap:?] {re:classloading}
        at de.bluecolored.shadow.benmanes.caffeine.cache.BoundedLocalCache.performCleanUp(BoundedLocalCache.java:1448) [bluemap:?] {re:classloading}
        at de.bluecolored.shadow.benmanes.caffeine.cache.BoundedLocalCache$PerformCleanupTask.run(BoundedLocalCache.java:3293) [bluemap:?] {re:classloading}
        at de.bluecolored.shadow.benmanes.caffeine.cache.BoundedLocalCache$PerformCleanupTask.exec(BoundedLocalCache.java:3280) [bluemap:?] {re:classloading}
        at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290) [?:?] {}
        at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020) [?:?] {}
        at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656) [?:?] {re:computing_frames}
        at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594) [?:?] {re:computing_frames}
        at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183) [?:?] {}

```

**Comment by TBlueF** [comment]
Should be finally fixed in 2.0 ... (as always: please comment/reopen if issue persists)
